### PR TITLE
Api 1270 copy flex asset types between accounts

### PR DESCRIPTION
**Pull request by lisath92**
This allows copying of all flexible asset types from one IT Glue account to another

### CODE REVIEW

File: Python Tooling Scripts/copy_flexible_asset_types.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    new_attribute = attributes
+    for key in EXCLUDE_KEYS:
+        new_attribute.pop(key, None)
+    return new_attribute
+
+
+def main():
+    source_api, target_api = get_api_url_and_key()
+    flex_types = get_all_flex_types(source_api)
+    type_id_maapping, tagged_list, tags_only_types = copy_flex_types(target_api, flex_types)
```
[the remark this answers is not in the document]
Comment (lisath92) [reply Python Tooling Scripts/copy_flexible_asset_types.py]: fixed

[the remark this answers is not in the document]
Comment (lisath92) [reply Python Tooling Scripts/copy_flexible_asset_types.py]: fixed

File: Python Tooling Scripts/copy_flexible_asset_types.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    * Created types in the new account with its corresponding new ID
+    * List of flexible asset fields with tag-type of another flexible asset types
+    * List of flexible asset types with only tag-types of "FlexibleAssetType"
+    """
+    fa_types_matching = {}
+    tag_fields = []
+    all_fields = get_all_fields(flex_types)
+    tags_only_types = []
+    for flex_type in flex_types:
+        type = find_or_initialize_flex_type(target_api_key, flex_type.attributes['name'])
```
[the remark this answers is not in the document]
Comment (lisath92) [reply Python Tooling Scripts/copy_flexible_asset_types.py]: changed all instances of `type` name to somethingelse

File: Python Tooling Scripts/copy_flexible_asset_types.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+def copy_flex_types(target_api_key, flex_types):
+    """Create flexible asset types that does not have a
+    flexible asset type tag field in the new account
+    Returns 3 lists:
+    * Created types in the new account with its corresponding new ID
+    * List of flexible asset fields with tag-type of another flexible asset types
+    * List of flexible asset types with only tag-types of "FlexibleAssetType"
+    """
+    fa_types_matching = {}
```
[the remark this answers is not in the document]
Comment (lisath92) [reply Python Tooling Scripts/copy_flexible_asset_types.py]: I've moved the set credentials after getting all the fields from the source account

File: Python Tooling Scripts/copy_flexible_asset_types.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        return ITGlueError('This flexible asset type {} does not have any fields.'.format(flex_type.attributes['name']))
+    for field in flex_asset_fields:
+        field_attr = extract_attributes(field.attributes)
+        flex_field = itglue.FlexibleAssetField(**field_attr)
+        updated_fields.append(flex_field)
+    return updated_fields
+
+
+def extract_attributes(attributes):
+    new_attribute = attributes
```
[the remark this answers is not in the document]
Comment (lisath92) [reply Python Tooling Scripts/copy_flexible_asset_types.py]: not a list, but a dictionary. However, I realized it doesn't make sense to make a copy here, removed this and using `attributes` directly

File: Python Tooling Scripts/copy_flexible_asset_types.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+def create_new_flex_types(api_key, new_type, new_fields):
+    set_connection_creds(api_key)
+    flex_name = new_type.attributes['name']
+    print('{} '.format(flex_name).ljust(50, '.'), end='')
+    try:
+        created_type = new_type.create(flexible_asset_fields=new_fields)
+        print('copied')
+        return created_type
+    except Exception as e:
+        print('please fix error')
```
[the remark this answers is not in the document]
Comment (lisath92) [reply Python Tooling Scripts/copy_flexible_asset_types.py]: nope, removed

File: Python Tooling Scripts/copy_flexible_asset_types.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    return type
+
+
+def create_new_flex_types(api_key, new_type, new_fields):
+    set_connection_creds(api_key)
+    flex_name = new_type.attributes['name']
+    print('{} '.format(flex_name).ljust(50, '.'), end='')
+    try:
+        created_type = new_type.create(flexible_asset_fields=new_fields)
+        print('copied')
```
[the remark this answers is not in the document]
Comment (lisath92) [reply Python Tooling Scripts/copy_flexible_asset_types.py]: Updated the print output

**Comment by itgmj** [comment]
Approved

**Pull request merged**